packages/winpanda/extra/src/winpanda/common/storage.py

```diff
@@ -0,0 +1,243 @@
+"""Panda package management for Windows.
+
+DC/OS installation local storage management tools.
+
+Default local storage layout for DC/OS installation:
+
+<inst_drive>:/                # DC/OS installation drive
+    +-<inst_root>/            # DC/OS installation root dir
```

Comment:
In the future, we are likely to want to position these folders in different locations. `<inst_tmp>` for example might map to a folder in `C:\Temp`. Similarly other folders might need to be on separate drives.

We should define default equivalents to:
INSTALL_FOLDER `/opt/mesosphere` -> `C:\dcos\opt`
VARLOG_FOLDER `/var/log/dcos` -> `C:\dcos\log`
VARLIB_FOLDER `/var/lib/dcos` -> `C:\dcos\lib`
... and anything else ...
without breaking them into drive and path components, and leave open the possibility of individually modifying each of these FOLDER variables through command line options.

Reply:
This is **default** FS layout. Every individual directory path can be easily redefined via an appropriate CLI option:
--inst-root-dir=<path>         DC/OS installation root directory
--inst-config-dir=<path>       DC/OS installation configuration directory
--inst-state-dir=<path>        DC/OS installation state directory
--inst-repo-dir=<path>         DC/OS local package repository directory.
--inst-var-data-dir=<path>     DC/OS variable data root directory.

Every of the options above accepts both relative and absolute pathes as their values.
Relative pathes will be resolved basing on the value of '--inst-root-dir' option. Absolute pathes will feature the drive part indeed and so we'll have enough flexibility to configure DC/OS installation local FS layout. I.e. this implementation already covers everything you are concerning about?